packages/core/contracts/zodiac/OptimisticGovernor.sol

```diff
@@ -0,0 +1,321 @@
+            }
+        }
+        proposal.transactions = _transactions;
+
+        // proposalHashes[id] = keccak256(abi.encodePacked(proposalData));
+        proposalHashes[id] = keccak256(abi.encode(_transactions));
+
+        // Get the bond from the proposer and approve the bond and final fee to be used by the oracle.
+        uint256 totalBond = finalFee + bond;
+        collateral.safeTransferFrom(msg.sender, address(this), totalBond);
+        collateral.safeIncreaseAllowance(address(skinnyOptimisticOracle), totalBond);
+
+        // Propose a set of transactions to the OO. If not disputed, they can be executed with executeProposal().
+        // docs: https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/oracle/interfaces/SkinnyOptimisticOracleInterface.sol
+        skinnyOptimisticOracle.requestAndProposePriceFor(
+            identifier,
+            uint32(time),
+            ancillaryData,
+            collateral,
+            0,
+            bond,
+            uint256(liveness),
+            address(this),
+            // Canonical value representing "True"; i.e. the transactions are valid.
+            int256(1e18)
+        );
+
+        emit TransactionsProposed(id, proposer, time, proposal);
+    }
+
+    function executeProposal(
+        uint256 _proposalId,
+        Transaction[] memory _transactions,
+        bytes memory _explanation,
+        uint32 _originalTime,
+        SkinnyOptimisticOracleInterface.Request memory _request
+    ) public payable nonReentrant {
+        // Recreate the proposal hash from the inputs and check that it matches the stored proposal hash.
+        uint256 id = _proposalId;
+
+        // Construct the ancillary data.
+        bytes memory ancillaryData = _explanation;
+        AncillaryData.appendKeyValueUint(ancillaryData, "id", id);
+
+        // This will reject the transaction if the proposal hash generated from the inputs does not match the stored proposal hash.
+        // require(proposalHashes[id] == keccak256(abi.encodePacked(proposalData)), "proposal hash does not match");
+        require(proposalHashes[id] == keccak256(abi.encode(_transactions)), "proposal hash does not match");
+
+        // Remove proposal hash so transactions can not be executed again.
+        proposalHashes[id] = 0;
+
+        // This will revert if the price has not settled.
+        (, int256 price) =
+            skinnyOptimisticOracle.settle(address(this), identifier, _originalTime, ancillaryData, _request);
+        require(price == 1e18, "Proposal was rejected");
+
+        for (uint256 i = 0; i < _transactions.length; i++) {
+            require(i == 0 || _transactions[i - 1].to == address(0), "Previous tx not yet executed");
+            Transaction memory transaction = _transactions[i];
+            require(transaction.to != address(0), "Tx already executed");
+            require(msg.value == transaction.value, "Must send exact amount of ETH");
```

Comment:
Since all transactions are being executed in the same call, I don't think this check makes sense, since there could be two calls that each use half the msg.value.

Do you envision a scenario where the proposer would need to send eth through this call?

Side note: can you explain how calls are supposed to happen? Aren't they supposed to be made _through_ the target (gnosis-safe)? This seems to allow _this contract_ to call out directly to other contracts? Is that what we want?